### ISSUE TITLE
Question on Pod communication with AKS

### ISSUE THREAD

**Issue by VenkateshSrini** [openfaas/faas-netes#146]
I have a function that calls the kubernets service in the same cluster. Through a function deployed in open Faas I'm calling the service using the external ip. The function is not responding at all. I'm attaching the function for your reference
The other service send email using send grid. I have tested that service independently fro post man it works without any issue.
please let me know what is the reason for the same.

## Expected Behaviour
The function should call the other Kubernetes container and return value

## Current Behaviour
The function times out
## Steps to Reproduce (for bugs)

1. Deploy the attached code 
2. create a container that will send out an email.
3. you may need to change the outgoing model class in the function to suit your email container
4. set get way in Yaml before deply
5. 
[mailerFunction.zip](https://github.com/openfaas/faas-netes/files/1742834/mailerFunction.zip)
Try hitting the function from openFaas UI or postman

## Context
It has completely stopped me from show casing the strength of OpenFass. Question is can openFass Relay http class to other containers in the same cluster and name space. Any addition setting needed in .yaml files?

## Your Environment

* Docker version `docker version` (e.g. Docker 17.0.05 ): 17.12.0-ce

* Are you using Docker Swarm or Kubernetes (FaaS-netes)? Kubernetes on AKS

* Operating System and version (e.g. Linux, Windows, MacOS): Windows

* Link to your project or a code example to reproduce issue:

**Comment by alexellis** [comment]
Derek add label: question

**Comment by alexellis** [comment]
Derek set title: Question on Pod communication with AKS

**Comment by alexellis** [comment]
Hi @VenkateshSrini we don't believe there is an issue with OpenFaaS here - it sounds like it may be a mis-configuration at the networking level.

OpenFaaS function pods can access Pods in other namespaces via HTTP - an example of this is the way a Pod can call the API Gateway and the queue worker we have which can call pods.

I'd suggest you try using DNS names to access services within your cluster or open an issue with the AKS team.

https://github.com/Azure/AKS

**Comment by VenkateshSrini** [comment]
@alexellis ,
No this is not an issue with azur AKS. When I started to understand openFaas a much deeper. I see that it is deployed as pod in K8s cluster. The pod was in status image pull back state. When I saw my docker hub I saw that the image was not there. 

So I repeated the steps 
`Faas-cli deploy -f file name`

This failed saying that it could find a dependency newtonsoft.json, which is a nuget package.

Then I did `dotnet publish -c release -o publish` This command was successful
Then again I did `faas-cli build -f filename.yml`. This time the build was successful.

Then I saw the Build folder getting created

Then I did `Faas-cli push -f maile.yml` I was getting a permission denied error. Then I again went through your trouble shooting guide. I saw that in the yaml file we need to add the repository name. Yes I had missed it. Though I did for another function for this function I had missed in. Then I again repeated the command and now it got pushed.
Now when I did `faas-cli deploy -f filename` the image got push to by cluser successfully.

Now every thing works perfectly. Thanks for your patientce
